generator/writeFile.js

```diff
@@ -59,27 +59,22 @@ function test () {
  */
 
 function splitObject (path,file) {
-  var src = path; // Location of object file
-  var obj = file // Object file name
-  var srclen= src.length;
-  if(src.charAt(srclen-1)!=='/') src = src + '/'; //path correction
-  var location = src + obj // location of object file
-  let ddata = fs.readFileSync(location)
+  if(path.charAt(path.length-1)!=='/') path = path + '/'; //path correction
+  let ddata = fs.readFileSync(path + file)
   let grocery = JSON.parse(ddata)
   var len = grocery.length // Object Length
   var foldername = file.slice(0,-5) ;
-  foldername = foldername+ "_elements";
+  foldername = foldername + "_elements";
   var fileName;
-  if (!fs.existsSync(src + foldername)) fs.mkdirSync(src + foldername) // Check if Folder exists or not.
+  if (!fs.existsSync(path + foldername)) fs.mkdirSync(path + foldername) // Check if Folder exists or not.
   for (var i = 0; i < len; i++) {
     let temp = JSON.stringify(grocery[i]);
-    if(typeof grocery.name === "undefined") fileName = i + '.json'; // if no name attrib then index will be file name
-    else fileName = grocery[i].name + '.json';
+    fileName = grocery[i].name + '.json';
     fileName = fileName.replace(/ /g, '_'); // Replace space with underscore
     fileName = fileName.toLowerCase();
     // Checking if file already exists or not
-    if (!fs.existsSync(src + foldername + '/' + fileName)) fs.writeFileSync(src + foldername + '/' + fileName, temp);
-    else console.log('File <' + fileName + '> Already Exists.');
+    if (!fs.existsSync(path + foldername + '/' + fileName))
+      fs.writeFileSync(path + foldername + '/' + fileName, temp);
```

Comment:
you removed else statement

Reply:
adding else statement increased complexity :(

Reply:
i have used writeFile() that is already defined.